packages/hash/frontend/src/components/layout/PageHeader/AccountDropdown.tsx

```diff
@@ -25,19 +24,27 @@ export const AccountDropdown: VoidFunctionComponent<AccountDropdownProps> = ({
   logout,
   user,
 }) => {
-  const theme = useTheme();
-
   const buttonRef = useRef(null);
 
   const [open, setOpen] = useState(false);
 
   const id = open ? "account-popover" : undefined;
 
+  const userEmail = useMemo(() => {
+    const primaryEmail = user.properties.emails.find((email) => email.primary);
+
+    if (primaryEmail) {
+      return primaryEmail.address;
+    }
+
+    return user.properties.emails[0].address;
```

Comment:
Although I'm aware every account will always have a primary email, this line is a good fallback and even satisfies the TS compiler to not return `undefined` instead.

Reply:
If I were writing this I'd throw an error, this is not a state we want to silently let happen - and an error might help us catch something in our tests.

Reply:
Makes sense.

Reply:
Done here: https://github.com/hashintel/hash/pull/397/commits/b4b958907ec615b340a0bbe0f00ae914c3561722